.github/workflows/ci.yml

```diff
@@ -0,0 +1,112 @@
+          - SHOP3::ROVERS-TESTS
+          - SHOP3::SHORT-PDDL-TESTS
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Checkout submodules
+      shell: bash
+      run: |
+        git submodule update --init --recursive
+
+    # Lisp setup copied from here: https://github.com/3b/ci-example/blob/master/.github/workflows/CI.yml
+    - name: cache .roswell
+      id: cache-dot-roswell
+      uses: actions/cache@v1
+      with:
+        path: ~/.roswell
+        key: ${{ runner.os }}-dot-roswell-${{ matrix.lisp }}-${{ hashFiles('**/*.asd') }}
+        restore-keys: |
+          ${{ runner.os }}-dot-roswell-${{ matrix.lisp }}-
+          ${{ runner.os }}-dot-roswell-
+
+    - name: install roswell
+      shell: bash
+      # always run install, since it does some global installs and setup that isn't cached
+      env:
+       LISP: ${{ matrix.lisp }}
+      # Use a previous release of Roswell to avoid error encountered
+      # due to libcurl3 not being available.
+      # Source of fix: https://github.com/avodonosov/drakma/commit/fbba29181ba2962f5031da581bd2de4dac98733d
+      run: |
+        sudo apt install -y libcurl4
+        curl -L https://raw.githubusercontent.com/roswell/roswell/a8fd8a3c33078d6f06e6cda9d099dcba6fbefcb7/scripts/install-for-ci.sh | sh
+
+    # Prompts Roswell to setup if the selected lisp version, if it has not been setup yet.
+    - name: run lisp
+      continue-on-error: true
+      shell: bash
+      run: |
+        ros -e '(format t "~a:~a on ~a~%...~%~%" (lisp-implementation-type) (lisp-implementation-version) (machine-type))'
+        ros -e '(format t " fixnum bits:~a~%" (integer-length most-positive-fixnum))'
+        ros -e "(ql:quickload 'trivial-features)" -e '(format t "features = ~s~%" *features*)'
+
+    # Compile first in a separate step to make the test output more readable
+    - name: compile
+      shell: bash
+      run: |
+        ros -e "(cl:in-package :cl-user)
+           (prin1 (lisp-implementation-type)) (terpri) (prin1 (lisp-implementation-version)) (terpri)
+           (prin1 \"${{ matrix.test }}\") (terpri)
+           (asdf:initialize-source-registry  '(:source-registry (:directory \"$PWD/shop3/\") (:tree \"$PWD/jenkins/ext/\") :inherit-configuration))
+           (declaim (optimize (speed 3)))
+           (asdf:load-system :shop3/test)
+           (uiop:quit 0)"
+
+    - name: tests
+      shell: bash
+      run: |
+        ros -e "(cl:in-package :cl-user)
+           (require :asdf)
+           (prin1 (lisp-implementation-type)) (terpri) (prin1 (lisp-implementation-version)) (terpri)
+           (prin1 \"${{ matrix.test }}\") (terpri)
+           (asdf:initialize-source-registry  '(:source-registry (:directory \"$PWD/shop3/\") (:tree \"$PWD/jenkins/ext/\") :inherit-configuration))
+           (declaim (optimize (speed 3)))
+           (asdf:load-system :shop3/test)
+           (defun test-fail-p (test-result)
+             (typecase test-result
```

Comment:
It looks like `fiveam::run-resolving-dependencies` might return `nil` for failing tests.  So maybe we need to refine `test-fail-p`.

Reply:
My understanding is that this will be caught by the `(null t)` case. I'm going to mark this resolved, but let me know if I misunderstood and this is separate from the `(null t)` case.